sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/state/SynchronousMetricStorage.java

```diff
@@ -154,30 +164,173 @@ public void recordDouble(double value, Attributes attributes, Context context) {
   }
 
   @Override
-  public MetricData collectAndReset(long startEpochNanos, long epochNanos) {
+  public MetricDescriptor getMetricDescriptor() {
+    return metricDescriptor;
+  }
+
+  /**
+   * This method leverages the `savedDeltas` and `reportHistory` to construct a final metric point
+   * to report for this synchronous instrument.
+   */
+  @GuardedBy("collectLock")
+  @Nullable
+  private MetricData buildMetricFor(
```

Comment:
let's move this method down below where it's called.

Reply:
also, what would you think about renaming it to `buildMetricsFor` ?

Reply:
I'm ok renaming....

Plurality here is hard.  It's one metric-data, but has multiple streams, meh.   The other question I have is should we move this to Collection<MetricData>?